crates/solana-event-forwarder/src/component.rs

```diff
@@ -53,38 +82,135 @@ impl SolanaEventForwarder {
         while let Some(message) = self.solana_listener_client.log_receiver.next().await {
             let gateway_program_stack =
                 build_program_event_stack(&match_context, &message.logs, parse_gateway_logs);
+            let gas_events_program_stack =
+                build_program_event_stack(&match_context, &message.logs, parse_gas_service_log);
+            // todo -- total cost is not representative
```

Comment:
The situation is the following:
- we require funds to approve messages, verify signatures, upload raw payload. This cost is spread out through many small transactions and the `total_cost` needs to be aggregated from all of this

I will mark this as a separate issue because it will just be extra bloat in this PR.